packages/dcos-integration-test/extra/resiliency/test_aws_cf.py

```diff
@@ -32,21 +33,20 @@ def dcos_launchpad(cluster):
 @pytest.mark.last
 def test_agent_failure(dcos_launchpad, cluster, vip_apps):
     # make sure the app works before starting
+    def get_running_agents(group_name):
```

Comment:
I think you're right. The issue with using resource objects instead of client objects for everything is that way boto is implemented, when you ask for these properties, it just calls out to client.describe_instances() with a specific filter. So each i.state would be a new client call. 

This is actually pretty crucial... the nice thing about keeping this the way it was is that you never need to wrap functions after you provide the instance info because it because the client calls are all made in safe method.  This unleashes the possibility of any dependent usage causing rate limit failures.

So, maybe we should wrap boto clients directly. I.E. patching at runtime. It will be a big ugly patch, but its the only way to allow the usage of these resource objects without taking a step backwards in CI stability